checker/values.mli

```diff
@@ -10,17 +10,44 @@
 
 type value =
   | Any
+  (** A value that we won't check, *)
+
   | Fail of string
+  (** A value that shouldn't be there at all, *)
+
   | Tuple of string * value array
+  (** A debug name and sub-values in this block *)
+
   | Sum of string * int * value array array
+  (** A debug name, a number of constant constructors, and sub-values
+     at each position of each possible constructed variant *)
+
   | Array of value
+  (** Ocaml array *)
```

Comment:
If you really think the information convoyed by this comment is relevant you can rename the type "ocaml_value" or the constructor OCamlArray. I'd drop all all the comment on Array, List, String, Opt and Int. Maybe put a single comment "OCaml containers and primitive data types" and move these constructors under it.

Reply:
Feel free to ignore this, but maybe next time don't write the comments in the first place.